### ISSUE TITLE
function declarations get converted to variable assignment without infinite loop safety

### ISSUE THREAD

**Issue by EricBalingit** [Khan/live-editor#748]
variable declaration expressions such as:
```js
var test1 = function ( a, b ) {
    return a + b;
};
```
result in the expected output with `KAInfiniteLoopProtect`

while function declarations such as
```js
function test2 ( a, b ) {
    return a + b;
}
```
do not

repro here:
https://www.khanacademy.org/computer-programming/func-decl-vdecl-without-loop-safety/5771192026742784

**Comment by pamelafox** [comment]
Yeah, I'm not surprised that there are issues like that. I added the function syntax for the AP CSP demos, but I don't use them anywhere in Intro to JS since I think there'd also be issues with the coding challenge auto-grader. It would need to be done in a more sophisticated manner than my quick approach. I'll leave this issue open. But generally, I would suggest folks sticks with the var syntax for the best chance of things working.

**Comment by lgto4** [comment]
I thought it was a feature that I take advantage of all the time.
